### DISCUSSION TITLE
PHP/Composer version requirements not enforced on composer install

### DISCUSSION THREAD

**Discussion by gms8994** [composer/composer#11989] [Q&A / Support] [answered]
As the title says, php and composer version requirements that are in the composer.json file aren't enforced when you run `composer install`.

Example in a new project:
`composer require psr/log`

Edit the composer.json and add to the `require` block `"php": "^10.0", "composer": "^3.0"`

Now, running `composer install` will not trigger failures, but you will get a message

```
Warning: The lock file is not up to date with the latest changes in composer.json. You may be getting outdated dependencies. It is recommended that you run `composer update` or `composer update <package name>`.
```

Is there a reason that this isn't being enforced? I'd like to provide a way to all the members of my team to say "we're upgrading the composer that is used to install things to this version" so that they'll know immediately that something might be different.

**Comment by Seldaek** [comment]
That's probably just an omission. I'd have to dig a bit deeper to really see why that is though.

**Reply by Seldaek** [reply, marked as answer]
Indeed looks like it was an omission because platform packages aren't part of the lock file we couldn't check them in the same way we check other dependencies. https://github.com/composer/composer/pull/12001 fixes it.
